packages/provider/wrapper.js

```diff
@@ -32,6 +32,43 @@ module.exports = {
       const originalSendAsync = provider.sendAsync.bind(provider);
       provider.sendAsync = this.send(originalSendAsync, preHook, postHook);
     }
+
+    // EIP-1193 provider wrapping
+    if (
+      typeof provider.request === "undefined" &&
+      typeof provider.send !== "undefined"
+    ) {
+      provider.request = async function ({ method, params }) {
+        let id = 0;
+        return await new Promise((accept, reject) => {
```

Comment:
Is there a reason we need to use `new Promise`? I prefer `async/await` syntax for async operations, to make the code more concise and easier to read. Perhaps something like: 

```
const response = await provider.send({
        jsonrpc: "2.0",
        id: ++requestId,
        method,
        params
      });
```

Reply:
I think it is because `response` would be `undefined` here since `provider.send` uses callbacks

Reply:
I agree with @davidmurdoch here. `provider.send` uses callbacks and we are wrapping this function in a callback style way to use `await` for the response/error.